hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java

```diff
@@ -88,7 +94,13 @@ public void execute(ScmClient scmClient) throws IOException {
       allNodes = allNodes.filter(p -> p.getHealthState().toString()
           .compareToIgnoreCase(nodeState) == 0);
     }
-    allNodes.forEach(this::printDatanodeInfo);
+
+    if (json) {
+      List<DatanodeWithAttributes> datanodeList = allNodes.collect(Collectors.toList());
+      System.out.print(JsonUtils.toJsonStringWithDefaultPrettyPrinter(datanodeList));
```

Comment:
Sorry, I just noticed that these lines are too long for checkstyle:

```
hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/ListInfoSubcommand.java
 99: Line is longer than 80 characters (found 88).
 100: Line is longer than 80 characters (found 85).
```

https://github.com/jyotirmoy-gh/ozone/actions/runs/4574046184/jobs/8075209766#step:6:511

Reply:
Reduced the line length